packages/account/src/Sections/Profile/PhoneVerification/resend-code-timer.tsx

```diff
@@ -4,41 +4,46 @@ import { Localize } from '@deriv/translations';
 
 type TResendCodeTimer = {
     resend_code_text: string;
+    count_from: number;
 };
-const ResendCodeTimer = ({ resend_code_text }: TResendCodeTimer) => {
+const ResendCodeTimer = ({ count_from = 60, resend_code_text }: TResendCodeTimer) => {
     // TODO: Use dynamic value for setting initial value for counter once mockApi for timestamp is finalised
-    const [timer, setTimer] = React.useState(60);
-    const [start_timer, setStartTimer] = React.useState(false);
-    const request_code = useRef(resend_code_text);
+    const [timer, setTimer] = React.useState(count_from);
+    const [start_timer, setStartTimer] = React.useState(true);
+    // TODO: change seconds to mins based on interval calculated using timestamp from mock api
+    const initial_timer_title =
+        resend_code_text === 'Resend code' ? `Resend code in ${timer}s` : `Didn’t get the code?(${timer}s)`;
+    const [timer_title, setTimerTitle] = React.useState(initial_timer_title);
+
+    const setTitle = (timer: number, text: string) => {
+        const title = text === 'Resend code' ? `Resend code in ${timer}s` : `Didn’t get the code?(${timer}s)`;
+        setTimerTitle(title);
+    };
 
     React.useEffect(() => {
         let countdown: ReturnType<typeof setInterval>;
         if (start_timer && timer > 0) {
             countdown = setInterval(() => {
                 setTimer(prevTime => prevTime - 1);
-                // TODO: change seconds to mins based on interval calculated using timestamp from mock api
-                request_code.current =
-                    resend_code_text === 'Resend code'
-                        ? `Resend code in ${timer - 1}s`
-                        : `Didn’t get the code?(${timer - 1}s)`;
+                setTitle(timer, resend_code_text);
             }, 1000);
         } else {
             setStartTimer(false);
-            request_code.current = resend_code_text;
+            setTimerTitle(resend_code_text);
         }
 
         return () => clearInterval(countdown);
     }, [timer, start_timer]);
 
     const resendCode = () => {
-        setTimer(60);
+        setTimer(count_from);
```

Comment:
Actually, the timer is not dependent on that BE response, it starts immediately and then we wait for some time just incase the user does not receive OTP then we enable the link.

Reply:
but is it expected to extend the timer for further attempts?
anyway I believe we should start timer only when the API call is successful, or how should we handle it

Reply:
The timer starts immediately when we come to this page, bcz its expected that user shld receive verification link on their email (not sure if there is a way to check that) so as a fallback when user finds they didn't receive OTP we re-enable this button so they can request for it(after timer expires). @amina-deriv ?

Reply:
Anyways, lets add a TODO since the flow for  resending verification link on email is missing in the design. For now we can request for verification only through OTP. 🙏